resources/assets/lib/chat/conversation-list.tsx

```diff
@@ -1,33 +1,47 @@
 // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
 // See the LICENCE file in the repository root for full licence text.
 
-import { inject, observer } from 'mobx-react';
-import Channel from 'models/chat/channel';
+import { computed } from 'mobx';
+import { observer } from 'mobx-react';
+import core from 'osu-core-singleton';
 import * as React from 'react';
 import ConversationListItem from './conversation-list-item';
 
-@inject('dataStore')
 @observer
-export default class ConversationList extends React.Component<any> {
+export default class ConversationList extends React.Component {
+  @computed
+  private get nonPmChannels() {
+    return core.dataStore.channelStore.nonPmChannels;
+  }
+
+  @computed
```

Comment:
are the `@computed` actually useful considering all the functions do is simply returning another `@computed` value?